app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt

```diff
@@ -407,6 +410,26 @@ class UploadWorker(var appContext: Context, workerParams: WorkerParameters) :
                     contribution.state = Contribution.STATE_FAILED
                     contribution.chunkInfo = null
                     contributionDao.saveSynchronous(contribution)
+
+                    //If the Current Login is Invalid then ask the User to Login Again
+                    if(stashUploadResult.message.equals("Invalid token, or login failure.")){
```

Comment:
I would avoid writing the string verbatim here. `"Invalid token, or login failure."` is repeated twice, here and at https://github.com/commons-app/apps-android-commons/blob/45c82b1a74c17e7bd36a26341fac112e23c339bc/app/src/main/java/fr/free/nrw/commons/auth/csrf/CsrfTokenClient.kt#L59 .

A quick fix would be defining a constant, or maybe having an enum for the failure causes (for now if tlooks like we only deal with one cause, though).